hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1417,9 +1414,16 @@ func (r *HostedClusterReconciler) reconcileIgnitionServer(ctx context.Context, h
 			},
 		}
 		hyperutil.SetColocation(hcluster, ignitionServerDeployment)
-		hyperutil.SetRestartAnnotation(hcluster, ignitionServerDeployment)
 		hyperutil.SetControlPlaneIsolation(hcluster, ignitionServerDeployment)
 		hyperutil.SetDefaultPriorityClass(ignitionServerDeployment)
+		switch hcluster.Spec.ControllerAvailabilityPolicy {
+		case hyperv1.HighlyAvailable:
+			ignitionServerDeployment.Spec.Replicas = k8sutilspointer.Int32Ptr(3)
```

Comment:
Need to have a rolling update strategy with 1 replica unavailable to progress deployment in 3 AZ situation